src/validation.cpp

```diff
@@ -5661,6 +5661,7 @@ void Chainstate::InvalidateCoinsDBOnDisk()
             "snapshot directory %s, otherwise you will encounter the same error again "
             "on the next startup.",
             src_str, dest_str, src_str));
+        throw;
```

Comment:
This seems like a significant behavior change. The way I read this, `InvalidateCoinsDBOnDisk` is only called by `MaybeCompleteSnapshotValidation`. It's return type is ignored by its call site in `ConnectTip`, which I think would now lead us to skip over some code afterwards.

Reply:
re: https://github.com/bitcoin/bitcoin/pull/27862#discussion_r1227196719

> This seems like a significant behavior change.

It does only happen when snapshot validation fails (so the node would be shutting down anyway), and renaming the snapshot also fails. Just reraising the exception seemed like this simplest possible approach to take (copied from code in `ValidatedSnapshotCleanup`). But it is simple enough to clean up the code more. I pushed a new version which gets rid of the AbortNode call and uses `util::Result` instead.

Reply:
Nice, this is exactly what I had in mind! Can be resolved.